### ISSUE TITLE
E-Mail Fetching After Upgrade

### ISSUE THREAD

**Issue by cstrock** [tiredofit/docker-freescout#51]
I have a Docker Container running FreeScout v 1.7.3 (So FreeScout v 1.5.2).  I've been trying to upgrade the container for a while, and whenever I do, the container stops downloading/importing new emails.  The Fetch E-Mail Command runs, but it doesn't detect any new emails. though it doesn't error and no emails show up in the help desks.  I can't find any answers in the log either.  (I'm using snapshots on a VM so I can always revert to my working version and it starts picking up the emails again.

Any ideas on where to look to find out why it's not picking up the emails after an upgrade?

**Comment by tiredofit** [comment]
Wow - yes that's quite a big jump in between versions. There are PHP changes, nginx changes, and also application changes that could be affecting this. I wonder, when you do the upgrade, have you tried to re-enter the passwords for the accounts/ldap? Secret hashing could have changed, although that should also show up somehow in the logs

I'd hate to have you jump from docker image to docker image, but that is always an option. Last resort I think it would take me 2 hours to get you upgraded through some fairly surgical means to get you back on the more recent track if you wanted to go down that path for personalized support.

**Comment by cstrock** [comment]
So I thought the same thing, so I jumped from 1.5.2 to the latest 1.5.x version and the same issue occurred.  I can try going version by version 1 at a time.  Worth a shot.  Just pulling new image is all I need to do right, to upgrade?  I shouldn't need to enter any console commands?

I actually have to re-enter the passwords after the upgrade every time, and it says "connection established". If it were a password issue, should it say something when I click the Fetch EMails tool?  It just says complete though.

Corey

**Comment by tiredofit** [comment]
I would try these commands if they aren't firing inside the container:

````
cd /www/freescout
sudo -u nginx php artisan freescout:after-app-update
sudo -u nginx php artisan migrate
sudo -u nginx php artisan freescout:clear-cache
sudo -u nginx php artisan config:clear
````

**Comment by cstrock** [comment]
I tried stepping through and I went through these versions:
1.7.3 Starting Version
1.7,4 Worked
1.7.5 Worked
1.7.6 Didn't exist
1.7.7 Worked
1.7.8 Worked
1.7.9 Not working  (And I tried the commands in the above message, just in case.)
1.7.10 Not working (Figured I'd try one more, just in case)

So the error starts on the change from FreeScout 1.5.2 -> 1.5.3  I guess I'll go check that changelog next.

**Comment by tiredofit** [comment]
Another way which would be pretty harsh would be to rename your .env / config file and then start up the container again letting it regenerate the config. You can copy your db credentials and APP_KEY in afterwards..
I remeber that time being especially messy with .env/config due to some changes with Nginx that were made on the Freescout end and I didn't fully understand it..

**Comment by cstrock** [comment]
I've been doing some more testing, I started with creating a new DB/etc and on 1.5.3+, it turns out that didn't work either.  So all this time, I was using Pop3 for this.  So i figured why not try IMAP instead.  And it turns out that Pop3 doesn't get the email, but IMAP does.  I've managed to upgrade my test version from 1.52 up to 1.79 using IMAP as the fetch method and it appears to be working.  So, I'm going to try switching the production one over to IMAP and upgrade it soon.  Thanks for the help.
